rms/search_engine/clients.py

```diff
@@ -0,0 +1,54 @@
+from abc import ABC, abstractmethod
+import httpx
+from rms.search_engine.models import SearchBody, ScopusSearchResponse
+from rms.settings import Settings
+
+
+class ScopusClient(ABC):
+
+    def __init__(self):
+        self.settings = Settings()
+        self.HEADERS = {
+            "httpAccept": "application/json",
+            "X-ELS-APIKey": self.settings.scopus_api_key
+        }
+        self.SCOPUS_SEARCH_API_ENDPOINT = self.settings.scopus_search_endpoint
+
+    @abstractmethod
+    def search(self, query: SearchBody) -> ScopusSearchResponse:
+        pass
+
```

Comment:
I would remove this as the search params and response types will vary depending on the endpoint